File: Editorial/89_Gray_Code/CSharp/3_Iteration.cs
public class Solution {
    public IList<int> GrayCode(int n) {
        IList<int> result = new List<int>();
        result.Add(0);
        for (int i = 1; i <= n; i++) {
            int previousSequenceLength = result.Count;
            int mask = 1 << (i - 1);
            for (int j = previousSequenceLength - 1; j >= 0; j--) {
                result.Add(mask + result[j]);
            }
        }

        return result;
    }
}

File: Editorial/89_Gray_Code/C/3_Iteration.c
int* grayCode(int n, int* returnSize) {
    int resultSize = 1;
    for (int i = 0; i < n; i++) {
        resultSize *= 2;
    }
    *returnSize = resultSize;
    int* result = (int*)malloc(sizeof(int) * resultSize);
    result[0] = 0;
    int length = 1;
    for (int i = 1; i <= n; i++) {
        int mask = 1 << (i - 1);
        for (int j = length - 1; j >= 0; j--) {
            result[length++] = mask + result[j];
        }
    }
    return result;
}
